frontend/src/metabase/models/containers/NewModelOptions/NewModelOptions.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useMemo } from "react";
+import { push } from "react-router-redux";
+import { t } from "ttag";
+import _ from "underscore";
+
+import { useMount } from "react-use";
+import { Location } from "history";
+import { Grid } from "metabase/components/Grid";
+import NewModelOption from "metabase/models/components/NewModelOption";
+
+import MetabaseSettings from "metabase/lib/settings";
+import * as Urls from "metabase/lib/urls";
+import Databases from "metabase/entities/databases";
+import { getHasDataAccess, getHasNativeWrite } from "metabase/selectors/data";
+
+import NoDatabasesEmptyState from "metabase/reference/databases/NoDatabasesEmptyState";
+
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { deserializeCardFromUrl } from "metabase/lib/card";
+import Database from "metabase-lib/metadata/Database";
+import {
+  OptionsGridItem,
+  OptionsRoot,
+  EducationalButton,
+} from "./NewModelOptions.styled";
+
+const EDUCATIONAL_LINK = MetabaseSettings.learnUrl("data-modeling/models");
+
+interface NewModelOptionsProps {
+  databases?: Database[];
+  location: Location;
+}
+
+const NewModelOptions = (props: NewModelOptionsProps) => {
+  const hasDataAccess = useSelector(() =>
+    getHasDataAccess(props.databases ?? []),
+  );
+  const hasNativeWrite = useSelector(() =>
+    getHasNativeWrite(props.databases ?? []),
+  );
+
+  const dispatch = useDispatch();
+
+  const collectionId = useMemo(() => {
+    const decodedCollectionHash = deserializeCardFromUrl(
+      location.hash.replace(/^#/, ""),
+    );
+    return decodedCollectionHash.collectionId;
+  }, []);
+
+  useMount(() => {
+    const { location } = props;
+    if (Object.keys(location.query).length > 0) {
```

Comment:
can't it be `location.search.length`?

Reply:
Yeah - I agree. I just converted the file to typescript so I didn't want to touch anything just yet, but if there's obvious improvements that won't affect the output then I'll add them 😄

Reply:
Hm... actually, looking at the code, I think it makes more sense to use `location.query` since we're accessing it in the next line. I think in a vacuum, `location.search.length` makes more sense but for consistency with the next few lines it might be easier for someone to come in and understand that we're checking the `query` length, then accessing it after